lib/internal/Magento/Framework/App/Filesystem/DirectoryResolver.php

```diff
@@ -44,12 +45,13 @@ public function __construct(DirectoryList $directoryList, Filesystem $filesystem
      *
      * @param string $path
      * @param string $directoryConfig
+     * @param string $driverCode
      * @return bool
      * @throws \Magento\Framework\Exception\FileSystemException
      */
-    public function validatePath($path, $directoryConfig = DirectoryList::MEDIA)
+    public function validatePath($path, $directoryConfig = DirectoryList::MEDIA, $driverCode = DriverPool::FILE)
```

Comment:
Unfortunately, we are not allowed to change public method signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

Reply:
I understand that I was aware of this rules. Basically what I wanted is to start a discussion about it and maybe we can find a solution. Since I was trying to implement a standalone file-system driver for cloud object storage like S3 without touching the local file-system (like 'database' file-system does) I found that even if it was engineered to be extendable and reusable there are very few touch-points like this where you should extend it to make it working. 

I don't see a way to fix it with backward compatibility in mind. It will be confusing to have multiple Directory Resolver  classes (even if one is deprecated) and also I could not find a name for a new method in this class and deprecate this one. 

Can you advise me with a best practice regarding this type of issues. I would really want to make this component fully extendable in core if it possible, than making it work in my module. 

Thank you!

Reply:
@georgebabarus, in this case, the optimal way to achieve described behavior is making this class configurable via DI. You may inject an optional `driverCode` parameter to the constructor and make `DriverPool::FILE` as default value. 

The best way is to introduce the new class with support driver code and deprecate current but you will need to refactor all usages and it may be a huge amount of affected files.

Reply:
@sidolov 
Thank you for advise. I manage to clarify my misunderstanding by reading readme file near DirectoryResolver. 
"The **\Magento\Framework\App\Filesystem** library extends **\Magento\Framework\Filesystem** by defining list of directories available in the application.
Refer directory codes from **\Magento\Framework\App\Filesystem\DirectoryList** class in **\Magento\Framework\Filesystem** operations to get access to available application directories."

What I understood from above is that this namespace is tightly-coupled to local file-system.
And this should be fine, but the confusion is that namespaces like Magento\Cms\Controller\Adminhtml\Wysiwyg are also tightly-coupled to the local file-system. 

So I guess we can add a better solution but it will took some time, so I will close this pull request. 

If Magento team has a plan for this component notify me, maybe I can help since I am working in this area.